Update language-specific docs to document that public functions can be called from the backend code (#5522)
slint-ui/slint: api/cpp/docs/generated_code.md

  * `invoke_<callback_name>` function which takes the callback argument as parameter and call the callback.
  * `on_<callback_name>` function which takes a functor as an argument and sets the callback handler
     for this callback. the functor must accept the type parameter of the callback

> * For each public function declared in the root component, an `invoke_<function_name>` function which
>   takes the function arguments as parameters and calls the function.

Reviewer: Suggestion for simplification. I think it's fair of users to expect that argument are passed along.

```suggestion
* For each public function declared in the root component, an `invoke_<function_name>` function to call the function.
```
